File: packages/graph/__fixtures__/extractor-project/validators.go
package validators

type User struct {
	Email    string `json:"email" validate:"required,email"`
	Name     string `json:"name" validate:"required,min=1,max=255"`
	Age      int    `json:"age" validate:"gte=0,lte=150"`
	Password string `json:"password" validate:"required,min=8"`
}

type Order struct {
	Amount   float64 `json:"amount" validate:"required,gt=0,lte=1000000"`
	Currency string  `json:"currency" validate:"required,oneof=USD EUR GBP"`
	Items    []string `json:"items" validate:"required,min=1"`
}

type Address struct {
	Street  string `json:"street" validate:"required,min=1"`
	City    string `json:"city" validate:"required,min=1"`
	ZipCode string `json:"zip_code" validate:"required"`
	Country string `json:"country" validate:"required,len=2"`
}

File: packages/graph/__fixtures__/extractor-project/Validators.java
import javax.validation.constraints.*;

public class User {
    @NotNull
    @Email
    private String email;

    @NotNull
    @Size(min = 1, max = 255)
    private String name;

    @Min(0)
    @Max(150)
    private int age;

    @NotNull
    @Size(min = 8)
    private String password;
}

public class Order {
    @NotNull
    @DecimalMin("0.01")
    @DecimalMax("1000000")
    private BigDecimal amount;

    @NotNull
    @Pattern(regexp = "^(USD|EUR|GBP)$")
    private String currency;

    @NotNull
    @Size(min = 1)
    private List<String> items;
}

public class Address {
    @NotNull
    @Size(min = 1)
    private String street;

    @NotNull
    @Size(min = 1)
    private String city;

    @NotNull
    @Pattern(regexp = "^\\d{5}(-\\d{4})?$")
    private String zipCode;

    @NotNull
    @Size(min = 2, max = 2)
    private String country;
}
